cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -209,16 +209,15 @@ void Run(const std::vector<string>& bag_filenames) {
     }
 
     bag.close();
-    // Ensure the clock is republished also during trajectory finalization,
-    // which might take a while.
-    clock_republish_timer.start();
     node.FinishTrajectory(trajectory_id);
-    clock_republish_timer.stop();
   }
 
-  // Republish the clock after bag processing has been completed.
+  // Ensure the clock is republished during the final optimization, which might
```

Comment:
This is not the only purpose, right? What about serialization and the `--keep_running` flag?